zio/jsonio/writer.go

```diff
@@ -1,26 +1,250 @@
 package jsonio
 
 import (
+	"bufio"
+	"bytes"
+	"encoding/hex"
 	"encoding/json"
+	"fmt"
 	"io"
+	"time"
 
 	"github.com/brimdata/zed"
+	"github.com/brimdata/zed/pkg/nano"
+	"github.com/brimdata/zed/pkg/terminal/color"
+	"github.com/brimdata/zed/zcode"
+	"github.com/brimdata/zed/zson"
+)
+
+var (
+	boolColor   = []byte("\x1b[1m")
+	fieldColor  = []byte("\x1b[34;1m")
+	nullColor   = []byte("\x1b[2m")
+	numberColor = []byte("\x1b[36m")
+	puncColor   = []byte{} // no color
+	stringColor = []byte("\x1b[32m")
 )
 
 type Writer struct {
 	io.Closer
-	encoder *json.Encoder
+	writer *bufio.Writer
+	tab    int
+
+	// Use json.Encoder to get marshal primitive Values. Have to use
+	// json.Encoder instead of json.Marshal because this is the only way to get
+	// pkg json to turn off HTML escaping.
```

Comment:
```suggestion
	// Use json.Encoder for primitive Values. Have to use
	// json.Encoder instead of json.Marshal because it's
	// the only way to turn off HTML escaping.
```